src/gui/mzroll/tabledockwidget.cpp

```diff
@@ -1158,6 +1158,32 @@ void TableDockWidget::deleteGroup(PeakGroup *groupX) {
   updateCompoundWidget();
 }
 
+bool TableDockWidget::deleteAllgroupsWarning() 
+{
+    QMessageBox *warning = new QMessageBox(this);
+    bool selectedOption;
+
+    auto htmlText = QString("<p><b>Are you sure you want to permanently erase all the "
+                            "groups from table?</b></p>");
+    htmlText += "<p>You can't undo this action.</p>";
+    warning->setText(htmlText);
+    warning->setIcon(QMessageBox::Icon::Warning);
+  
+    auto noButton = warning->addButton(tr("No"),
+                                  QMessageBox::RejectRole);
+    auto yesButton = warning->addButton(tr("Yes"),
+                                  QMessageBox::AcceptRole);
+    warning->exec();
+
+    if(warning->clickedButton() == yesButton)
+        selectedOption = true;
+    else
+        selectedOption = false;
+
+    QCoreApplication::processEvents();
```

Comment:
Why was this needed? If not for this, there would be no need for the `selectedOption` boolean and lines 1178 to 1184 can be reduced to:
```C++
if (warning->clickedButton() == yesButton)
    return true;

return false;
```